portfolio/src/main/webapp/js/custom-script.js

```diff
@@ -125,6 +125,91 @@ function populateDropdown(list, ID) {
   });
 }
 
+/**
+ * Places pins on the map in a given radius set by the user
+ */
+async function placeProximityPins() {
+  const state = document.getElementById('state').value;
+  const userAddress = document.getElementById('address').value;
+  const response = await fetch('/proximity-pins?state=' + state);
+  const pins = await response.json();
+  const radius = Number(document.getElementById('radius').value);
+
+  pins.forEach(async (pin) => {
+    if (await haversineDistance(userAddress, pin.address) < radius) {
+      codeAddress(pin.address);
+    }
+  });
+}
+
+/**
+ * Calculates the distance between two coordinate sets in miles
+ * @param {String} userAddress as the point to calculate the distance from
+ * @param {String} dataPoint as the point to calculate the distance between
+ * @return {number} Distance in miles.
+ */
+async function haversineDistance(userAddress, dataPoint) {
```

Comment:
Going to assume you double checked this formula because I don't know it and I'm not going to look it up :p 

Nice, descriptive variable names!

Reply:
yea i've done a good amount of research on it. (For future reference if you have any questions about it, [this](https://www.movable-type.co.uk/scripts/latlong.html) is a helpful link.)